In [4]:
import pandas as pd
from sklearn.model_selection import train_test_split

SEED = 42

train_df = pd.read_csv('../data/train.csv')

X = train_df.drop(columns=['readmitted_30_days'])
y = train_df['readmitted_30_days']


In [5]:
X_train, X_val, y_train, y_val = train_test_split(X, y, test_size=0.2, stratify=y, random_state=SEED)

X_train.shape

(24000, 14)

In [10]:
X_val.shape

(6000, 14)

In [22]:
y_train.value_counts(normalize=True)

readmitted_30_days
0    0.802583
1    0.197417
Name: proportion, dtype: float64

In [21]:
y_val.value_counts(normalize=True)

readmitted_30_days
0    0.8025
1    0.1975
Name: proportion, dtype: float64

## Train/Val Split - Findings
- 80/20 split
- Train: ~80.26% / ~19.74%
- Val: ~80.25% / ~19.75%
- Balance preserved as expected
